packages/template-retail-react-app/app/pages/cart/index.test.js

```diff
@@ -1229,85 +1230,51 @@ describe('Product bundles', () => {
     })
 })
 
-describe('Unavailable products tests', function () {
-    test('Remove unavailable/out of stock/low stock products from cart', async () => {
-        prependHandlersToServer([
-            {path: '*/customers/:customerId/baskets', res: () => mockBaskets},
-            {path: '*/products', res: () => mockProducts}
-        ])
+describe('Bonus products', () => {
+    beforeEach(() => {
+        global.server.use(
+            rest.get('*/customers/:customerId/baskets', (req, res, ctx) => {
+                return res(ctx.delay(0), ctx.json(mockBonusProductBasket))
+            })
+        )
+    })
 
-        const {user, getByText} = renderWithProviders(<Cart />)
+    test('renders bonus products in cart with correct styling and no quantity picker', async () => {
+        renderWithProviders(<Cart />)
+
+        // Wait for the cart to load
         await waitFor(() => {
-            expect(screen.getByTestId('sf-cart-container')).toBeInTheDocument()
-            expect(screen.getByText(/Worn Gold Dangle Earring/i)).toBeInTheDocument()
-            expect(screen.getByText(/Straight Leg Trousers/i)).toBeInTheDocument()
+            expect(screen.queryByTestId('sf-cart-skeleton')).not.toBeInTheDocument()
         })
 
-        await waitFor(async () => {
-            expect(getByText(/Items Unavailable/i)).toBeVisible()
-            expect(
-                getByText(
-                    /Some items are no longer available online and will be removed from your cart./i
-                )
-            ).toBeVisible()
-        })
-        await waitFor(async () => {
-            expect(getByText(/Items Unavailable/i)).toBeVisible()
-            expect(
-                getByText(
-                    /Some items are no longer available online and will be removed from your cart./i
-                )
-            ).toBeVisible()
-        })
+        const regularProduct = screen.getByTestId('sf-cart-item-701642889830M')
+        const bonusProduct = screen.getByTestId('sf-cart-item-013742335262M')
```

Comment:
Why don't we look for product name being render on the UI? From react-testing-library best practise, it is better to test using getByText/getByLabelText instead of relying on getByTestId